### PR TITLE
Step 13: Add status to make it searchable

### PR DESCRIPTION
**Pull request by stakahashi-d**
## Tasks
- Add status (not started / in progress / completed)
- Make it possible to search by title and status on the list screen
- Add on the search index on the table
- Add a model spec to the search (let's expand the system spec as well)

### Model
- Task can be reserved even status is empty
- Users can only put (0, 1, 2) in status

### System
- Users can search by status
- Put the text then users can seach tasks(target column: title, deescription)

### index page
- before
Screen Shot 2022-05-23 at 15 09 53

- after
Add search bar
Add status column to table 
I also made it to able to search in not only title but also description field
Screen Shot 2022-05-23 at 15 11 17

Screen Shot 2022-05-23 at 15 13 48

### edit page
before
Screen Shot 2022-05-20 at 12 23 18

after
add status field, so that user can chage status of task
Screen Shot 2022-05-23 at 15 14 26

### CODE REVIEW

File: myapp/spec/system/tasks_spec.rb [hunk 1]
```diff
@@ -4,57 +4,88 @@
...
-      it '正常に遷移すること' do
-        click_on 'タスク新規追加'
-        expect(current_path).to eq new_task_path
+      context '削除ボタンが押された時' do
+        it '削除が正常に行われること' do
+          all('table tr')[1].click_on '削除'
+          expect(page).to have_no_content first_task.description
+        end
       end
     end
```
Comment (stakahashi-d) [inline myapp/spec/system/tasks_spec.rb:60]: here is only indent change.
I havent't changed the operation

https://github.com/Fablic/training/pull/1308/files#diff-f8ed3cee4a450658bdd597575135d60c8a9fbdffd62ddffcaa978dc1ff1dd7bcL7-R60

File: myapp/spec/system/tasks_spec.rb [hunk 2]
```diff
@@ -4,57 +4,88 @@
...
 
-    context '編集ボタンが押された時' do
-      it '正常に遷移すること' do
-        all('table tr')[1].click_on '編集'
-        expect(current_path).to eq edit_task_path first_task.id
+    describe '検索機能' do
```
Comment (stakahashi-d) [inline myapp/spec/system/tasks_spec.rb:62]: I add the test of search function.
Which tests about status search, and title or description search

**Comment by jeff-yoo-rakuma** [comment]
[NIT] It will be nice to have some basic info like what it is, what it does, how to run and etc in README.md

**Comment by jeff-yoo-rakuma** [comment]
I think you'd created gitignore file after you've pushed all the files that should have been ignored. Please delete them and push the change again!

**Comment by stakahashi-d** [comment]
> I think you'd created gitignore file after you've pushed all the files that should have been ignored. Please delete them and push the change again!

Thank you for letting me know.
I made another PR because number of file was huge.

https://github.com/Fablic/training/pull/1311

**Comment by stakahashi-d** [comment]
> I think you'd created gitignore file after you've pushed all the files that should have been ignored. Please delete them and push the change again!

Thank you for letting me know.
I deleted them in my main branch! And applied changes to this branch

**Comment by stakahashi-d** [comment]
> [NIT] It will be nice to have some basic info like what it is, what it does, how to run and etc in README.md

Thank you for pointing out.
I added 'what function does application has' and 'how to set up application' to README file!
https://github.com/Fablic/training/pull/1308/files#diff-3315def260651dc67c2478cabcc1c5d6ec96aecbd2d389ecf0206fe657430515L1

File: myapp/app/controllers/tasks_controller.rb [hunk 3]
```diff
@@ -50,4 +54,13 @@ def sort_type
   def sort_column
     Task.column_names.include?(params[:sort]) ? params[:sort] : 'created_at'
   end
+
+  def tasks
+    if params[:search_text].present? || params[:search_status].present?
+      Task.where('(title like ? or description like ?) and status = ?', "%#{params[:search_text]}%",
+                 "%#{params[:search_text]}%", params[:search_status]).order("#{sort_column} #{sort_type}")
```
Comment (jeff-yoo-rakuma) [inline myapp/app/controllers/tasks_controller.rb:62]: [NIT] Showing some message when there's no search result could be nice.

**Review by jeff-yoo-rakuma — APPROVED** [review]
Good system specs!
Good job 👍

**Pull request merged**